### ISSUE TITLE
Font is always black after installing macOS 11 (Source Code Pro font or derivatives need to be updated)

### ISSUE THREAD

**Issue by anukul** [microsoft/vscode#100840]
- VSCode Version: 1.46.1
- OS Version: macOS Big Sur Version 11.0 Beta (20A4299v)

Steps to Reproduce:

1. Upgrade to macOS 11.0 Developer Beta 1
2. Open VS Code, switch between the bundled colour themes

<img width="846" alt="image" src="https://user-images.githubusercontent.com/44864521/85398498-f46e7400-b572-11ea-9b83-1f62a80cf7ec.png">

Does this issue occur when all extensions are disabled?: Yes

**Comment by beratgashi** [comment]
Same problem here :/

**Comment by AugustDev** [comment]
Same problem, unworkable.

**Comment by aeschli** [comment]
Likely a Electron/Chromium issue...

**Comment by misolori** [comment]
Not sure if this helps but I also upgraded to Big Sur and don't see this happening (either on Stable or Insiders):

<img width="701" alt="image" src="https://user-images.githubusercontent.com/35271042/85616658-eff99800-b612-11ea-8429-1d2aaaa58f3f.png">

**Comment by adregia** [comment]
For what it's worth, the issue subsided for me when I disabled my custom font settings.

With custom font enabled: 
<img width="1048" alt="Screen Shot 2020-06-24 at 1 12 43 PM" src="https://user-images.githubusercontent.com/12516271/85622948-7bc3f200-b61c-11ea-9a85-56927f9e0531.png">

Without:
<img width="1048" alt="Screen Shot 2020-06-24 at 1 12 56 PM" src="https://user-images.githubusercontent.com/12516271/85622966-82eb0000-b61c-11ea-81bb-c03fdae9717c.png">

The issue appears to manifest itself with Hasklig in particular. As you can see, Verdana is not problematic (also tried Helvetica):
<img width="1048" alt="Screen Shot 2020-06-24 at 1 15 42 PM" src="https://user-images.githubusercontent.com/12516271/85623190-e117e300-b61c-11ea-9bb6-cb19bcc22694.png">

Although Hasklig is not a pre-installed system font, I also tried Lexend Deca, which is also custom, and presented no issues rendering color. Reinstalling Hasklig did not alleviate the issue.

Scopes inspector indicates the correct foreground color:
<img width="1048" alt="Screen Shot 2020-06-24 at 1 20 02 PM" src="https://user-images.githubusercontent.com/12516271/85623565-73b88200-b61d-11ea-939d-096975a3f190.png">

**Comment by misolori** [comment]
@adregia thanks for investigating, can also confirm that using Hasklig breaks syntax highlighting. Other custom fonts like Input Mono seem to work fine though.

**Comment by alexdima** [comment]
@misolori I think this might be an electron/chromium issue. Could you please try the standalone editor in Chrome -- https://microsoft.github.io/monaco-editor/ . You could just edit the font-family on the editor `view-lines` dom node.

**Comment by misolori** [comment]
@alexdima also repros on the standalone editor in Chrome:

![image](https://user-images.githubusercontent.com/35271042/86024303-abd71080-b9e1-11ea-8724-ef1f646c8955.png)

**Comment by alexdima** [comment]
It looks like this font does not want to be rendered with colors in Big Sur by Chromium. 

@misolori Could you please create a simple HTML page and report an issue to Chromium at https://bugs.chromium.org/p/chromium/issues/list

**Comment by misolori** [comment]
Created https://bugs.chromium.org/p/chromium/issues/detail?id=1100502. Also worth noting that I tested this on Safari, Firefox, and Edge and they all fail to render the text except for Safari.

**Comment by alexdima** [comment]
Thanks a lot @misolori !

**Comment by jeremyFreeAgent** [comment]
I can confirm Hasklig breaks colors. Same issue on Hyper (https://github.com/vercel/hyper/).

**Comment by zavan** [comment]
`"editor.fontFamily": "Source Code Pro"` breaks it too, same problem. Removing the setting works.

**Comment by balupton** [comment]
Is font ligatures to blame then?

**Comment by Andesome** [comment]
disable the editor.fontFamily. i am working.

**Comment by seaneking** [comment]
This is down to a Source Code Pro rendering bug in chromium. I have no idea why, but anywhere that uses it or it's derivatives, including websites etc, has the same issue.

**Comment by falcn** [comment]
> The issue appears to manifest itself with Hasklig in particular. 
For anyone who needs replacement, ligaturized Source Code Pro is the same font as Hasklig, and it works with Big Sur
https://github.com/ToxicFrog/Ligaturizer/

**Comment by WillBeebe** [comment]
Downloading other font families you can easily repo this in OX11

W/ Monoid, it works, without it doesn't. Same w/ `Hasklig`
```
"Monoid", "Source Code Pro", sans-serif
```

**Comment by leepowelldev** [comment]
Is there anything close to Source Code Pro which is known to work?

**Comment by theLMGN** [comment]
It works in Visual Studio & Xcode, but not Firefox or Chromium
![image](https://user-images.githubusercontent.com/12257112/99071206-938b1b00-25a9-11eb-8f55-ae2658ed70f9.png)

**Comment by levrik** [comment]
Source Code Pro downloaded from https://github.com/adobe-fonts/source-code-pro/releases/tag/2.030R-ro%2F1.050R-it is broken.
Source Code Pro downloaded from https://fonts.google.com/specimen/Source+Code+Pro works completely fine but the font sizing is different compared to the one from GitHub.

**Comment by draganjovanovic1** [comment]
I can confirm @levrik's solution worked for me. Thanks

**Comment by jinmu333** [comment]
Source Code Pro from google works completely fine.

**Comment by leepowelldev** [comment]
Thank you @levrik 😆 😆 😆

**Comment by KishiTheMechanic** [comment]
Thank you very much @levrik 🙌

**Comment by joedborg** [comment]
@levrik you win the internet today

**Comment by balupton** [comment]
Please click the 👍 button on his comment to signal your appreciation to the commenter, rather than notifying everyone subscribed this thread which is distracting for them and the maintainers.

For the future, if there isn't a 👍 button on their comment, you can add it via the emoji button on the top right of their comment.

<img width="209" alt="Screen Shot 2020-11-14 at 12 59 04 pm" src="https://user-images.githubusercontent.com/61148/99136524-745bbe80-2679-11eb-922a-905631dc11ee.png">

**Comment by sogud** [comment]
I fixed the problem by reinstalling the fonts and restarting macOS.
[https://fonts.google.com/specimen/Source+Code+Pro](https://fonts.google.com/specimen/Source+Code+Pro)

**Comment by alexdima** [comment]
Also happens for `Hasklig` and `Programma` (from https://github.com/microsoft/vscode/issues/110640)

**Comment by 1moeses** [comment]
Reinstalled `Hasklig` , restarted macOS and the Issue still persists. All the other fonts are working fine for me now. Temporarily switched to `JetBrain Mono` but I already miss my good old `Hasklig`.

**Comment by alexdima** [comment]
`Hasklig` is forked from `Source Code Pro`. Upstream issue for `Hasklig` -- https://github.com/i-tu/Hasklig/issues/123

**Comment by thedocbwarren** [comment]
May be a ligature problem.  Works with font's that don't seem to support it.

**Comment by theLMGN** [comment]
I read somewhere that it’s emoji support, because emojis have Color, they don’t Color the rest of the character’s

**Comment by thedocbwarren** [comment]
Maybe, it could be both as in extended characters are an issue.

**Comment by vitahlin** [comment]
Same issue.
macOS version: macOS Big Sur 11.0.1
Visual Studio Code version: 1.51.1
I use Solarized Dark theme and my font is Source Code Pro, I reinstalled Source Code Pro(ttf) from https://github.com/adobe-fonts/source-code-pro and it fixed.

**Comment by reneruiz** [comment]
Same issue, fixed by removing Haskig off font stack. Not sure why it was added there over my own choice to begin with.

**Comment by TerrenceLJones** [comment]
> UPDATE: Just downloaded Source Code Pro again from Google Fonts, and overwrote my existing files, seems to work ok now.

This fixed the issue for me.

**Comment by alexdima** [comment]
This is also a problem for `Source Han Code JP`, "which is a derivative of `Source Han Sans` that replaces its proportional Latin glyphs with fixed-width 667-unit glyphs from `Source Code Pro`". (found via #110552)

**Comment by punchcutter** [comment]
The latest releases of [Source Code Pro](https://github.com/adobe-fonts/source-code-pro/releases/tag/2.032R-ro%2F1.052R-it%2F1.012R-VAR) and [Source Han Code JP](https://github.com/adobe-fonts/source-han-code-jp/releases/tag/2.012R) have removed the SVG table to make them work as expected on macOS 11.

**Comment by leepowelldev** [comment]
Does anyone get a collapse on certain html elements/attributes which end in 'l' and the attribute starts with 'l' ?? Screenshot attached showing how it behaves on `<html lang="en">` - it seems to collapse the whitespace and replace it with a dot.

<img width="287" alt="Screenshot 2020-11-25 at 09 17 49" src="https://user-images.githubusercontent.com/602052/100207579-a71a7800-2eff-11eb-9d3e-6cd121a8d11a.png">

**Comment by alexdima** [comment]
@leepowelldev Yes, I've seen this before. See #106583. The workaround is to use `"editor.fontLigatures": "'ccmp' off"`

**Comment by leepowelldev** [comment]
@alexdima Thank you so much! You have no idea how much that's been driving me nuts!

**Comment by arjunguha** [comment]
Very helpful. Thanks. I'm using Source Code Pro Version 2.030;PS 1.0;hotconv 16.6.51;makeotf.lib2.5.65220 and MacOS 11.0.1. Setting `"editor.fontLigatures": "'ccmp' off"` did *not* work for me.

However, removing 'Source Code Pro' from `"editor.fontFamily"` resolves the problem.

**Comment by gongteng0215** [comment]
> Is there anything close to Source Code Pro which is known to work?
> UPDATE: Just downloaded Source Code Pro again from Google Fonts, and overwrote my existing files, seems to work ok now.
It works for me. ts

**Comment by hoff2** [comment]
> `Hasklig` is forked from `Source Code Pro`. Upstream issue for `Hasklig` -- [i-tu/Hasklig#123](https://github.com/i-tu/Hasklig/issues/123)

I'm seeing it with Source Code Pro too so can confirm

**Comment by hidekuro** [comment]
Try re-install font to resolve problem.

In my case, I using "Source Han Code JP".
I saw correct colors after re-install the font.

**Comment by bjesuiter** [comment]
> For what it's worth, the issue subsided for me when I disabled my custom font settings.
> 
> With custom font enabled:
> <img alt="Screen Shot 2020-06-24 at 1 12 43 PM" width="1048" src="https://user-images.githubusercontent.com/12516271/85622948-7bc3f200-b61c-11ea-9a85-56927f9e0531.png">
> 
> Without:
> <img alt="Screen Shot 2020-06-24 at 1 12 56 PM" width="1048" src="https://user-images.githubusercontent.com/12516271/85622966-82eb0000-b61c-11ea-81bb-c03fdae9717c.png">
> 
> The issue appears to manifest itself with Hasklig in particular. As you can see, Verdana is not problematic (also tried Helvetica):
> <img alt="Screen Shot 2020-06-24 at 1 15 42 PM" width="1048" src="https://user-images.githubusercontent.com/12516271/85623190-e117e300-b61c-11ea-9bb6-cb19bcc22694.png">
> 
> Although Hasklig is not a pre-installed system font, I also tried Lexend Deca, which is also custom, and presented no issues rendering color. Reinstalling Hasklig did not alleviate the issue.
> 
> Scopes inspector indicates the correct foreground color:
> <img alt="Screen Shot 2020-06-24 at 1 20 02 PM" width="1048" src="https://user-images.githubusercontent.com/12516271/85623565-73b88200-b61d-11ea-939d-096975a3f190.png">

Thanks for your research, this works great! 
I normally use  'Source Code Pro' font, but when I delete it from my font settings, the next font will be used. 
The next font is 'Menlo' and since it's installed on my system, it loads just fine.

**Comment by LyonUp** [comment]
> `"editor.fontFamily": "Source Code Pro"` breaks it too, same problem. Removing the setting works.

I also solved it by installing the "Source Code Pro" font：https://github.com/adobe-fonts/source-code-pro/releases/tag/variable-fonts

**Comment by jundl77** [comment]
FYI, the fix for Hasklig is to install 1.2 or later: https://github.com/i-tu/Hasklig/releases/tag/v1.2

**Comment by Somesa** [comment]
Reinstalling 'Source Code Pro' font from google fonts worked for me
Thanks 👍

**Comment by guylyons** [comment]
Removing my font choice specification in the settings.json fixed it for me.

**Comment by behemos** [comment]
> Is there anything close to Source Code Pro which is known to work?
> UPDATE: Just downloaded Source Code Pro again from Google Fonts, and overwrote my existing files, seems to work ok now.

👍  https://fonts.google.com/specimen/Source+Code+Pro

**Comment by behemos** [comment]
> Is there anything close to Source Code Pro which is known to work?
> UPDATE: Just downloaded Source Code Pro again from Google Fonts, and overwrote my existing files, seems to work ok now.

👍  https://fonts.google.com/specimen/Source+Code+Pro

**Comment by blakewatson** [comment]
In case some poor soul has the same problem I did... Source Code Pro was all black in VS Code. No highlighting. I tried re-downloading a newer version of Source Code Pro as suggested above but it didn’t work.

After nearly going mad, I realized that I also had Source Code Pro installed via the Adobe Creative Cloud app and _that_ version of the font was the old broken one. VS Code was preferring the Adobe version. After removing that from Adobe CC and re-installing the latest version of Source Code Pro (and reloading the VS Code window) it works perfectly.
